a10_octavia/common/utils.py

```diff
@@ -401,28 +401,28 @@ def get_ipv6_address(ifnum_oper, subnet, nics, address_list, loadbalancers_list)
         if port_mac_address == acos_mac_address:
             if nic.network_id in amp_boot_networks:
                 final_address_list = get_ipv6_address_from_conf(address_list, nic.network_id)
-                if len(final_address_list) > 0:
+                if final_address_list and len(final_address_list) > 0:
                     break
 
             if nic.network_id == subnet.network_id and not loadbalancers_list:
                 final_address_list = get_ipv6_address_from_conf(address_list, subnet.id)
-                if len(final_address_list) > 0:
+                if final_address_list and len(final_address_list) > 0:
                     break
 
             if loadbalancers_list:
                 for lb in loadbalancers_list:
                     if lb.vip.network_id == nic.network_id:
                         final_address_list = get_ipv6_address_from_conf(address_list,
                                                                         lb.vip.subnet_id)
-                        if len(final_address_list) > 0:
+                        if final_address_list and len(final_address_list) > 0:
```

Comment:
No need to check if `final_address_list` exists here as it's being checked in the `get_ipv6_address_from_conf()` and if it is None then an exception is raised there.
https://github.com/a10networks/a10-octavia/pull/597/files

Please take latest pull to get those changes in this patch.

Reply:
removed it